app/src/main/java/org/schabi/newpipe/player/resolver/PlaybackResolver.java

```diff
@@ -68,38 +89,217 @@ default MediaSource buildLiveMediaSource(@NonNull final PlayerDataSource dataSou
+            throw new IOException("Error when parsing manual DASH manifest", e);
+        }
+
+        return dataSource.getDashMediaSourceFactory().createMediaSource(dashManifest,
                 new MediaItem.Builder()
-                    .setTag(metadata)
-                    .setUri(uri)
-                    .setCustomCacheKey(cacheKey)
-                    .build()
-        );
+                        .setTag(metadata)
+                        .setUri(Uri.parse(stream.getContent()))
+                        .setCustomCacheKey(cacheKey)
+                        .build());
+    }
+
+    default <T extends Stream> HlsMediaSource buildHlsMediaSource(
+            @NonNull final PlayerDataSource dataSource,
+            @NonNull final T stream,
+            @NonNull final String cacheKey,
+            @NonNull final MediaSourceTag metadata) throws IOException {
+        final boolean isUrlStream = stream.isUrl();
+        if (isUrlStream && isNullOrEmpty(stream.getContent())) {
+            throw new IOException("Try to generate a DASH media source from an empty string or "
+                    + "from a null object");
+        }
+        if (isUrlStream) {
+            return dataSource.getHlsMediaSourceFactory(null).createMediaSource(
+                    new MediaItem.Builder()
+                            .setTag(metadata)
+                            .setUri(Uri.parse(stream.getContent()))
+                            .setCustomCacheKey(cacheKey)
+                            .build());
+        } else {
+            String baseUrl = stream.getBaseUrl();
+            if (baseUrl == null) {
+                baseUrl = "";
+            }
+
+            final Uri uri = Uri.parse(baseUrl);
+
+            final HlsPlaylist hlsPlaylist;
+            try {
+                final ByteArrayInputStream hlsManifestInput = new ByteArrayInputStream(
+                        stream.getContent().getBytes(StandardCharsets.UTF_8));
+                hlsPlaylist = new HlsPlaylistParser().parse(uri, hlsManifestInput);
+            } catch (final IOException e) {
+                throw new IOException("Error when parsing manual HLS manifest", e);
+            }
+
+            return dataSource.getHlsMediaSourceFactory(
+                    new NewPipeHlsPlaylistParserFactory(hlsPlaylist))
+                    .createMediaSource(new MediaItem.Builder()
+                            .setTag(metadata)
+                            .setUri(Uri.parse(stream.getContent()))
+                            .setCustomCacheKey(cacheKey)
+                            .build());
+        }
+    }
+
+    /**
+     * Remove streams which are using the {@link DeliveryMethod#TORRENT torrent delivery method}.
+     *
+     * @param streamList the list of {@link Stream streams} for which you want to delete the
+     *                   torrent streams.
+     */
+    default void removeTorrentStreams(@NonNull final List<? extends Stream> streamList) {
```

Comment:
Why is this here and not in `ListHelper` like the others? And why is it not a `static` method?